File: dmn-test-cases/tck/1.1/cl3/0031-user-defined-functions/interpreter/expected/java/dmn/type/TFnLibraryImpl.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinition.ftl", "tFnLibrary"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public class TFnLibraryImpl implements TFnLibrary {
        private Object sumFn;
        private Object subFn;
        private Object multiplyFn;
        private Object divideFn;

    public TFnLibraryImpl() {
    }

    public TFnLibraryImpl(Object divideFn, Object multiplyFn, Object subFn, Object sumFn) {
        this.setDivideFn(divideFn);
        this.setMultiplyFn(multiplyFn);
        this.setSubFn(subFn);
        this.setSumFn(sumFn);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("divideFn")
    public Object getDivideFn() {
        return this.divideFn;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("divideFn")
    public void setDivideFn(Object divideFn) {
        this.divideFn = divideFn;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("multiplyFn")
    public Object getMultiplyFn() {
        return this.multiplyFn;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("multiplyFn")
    public void setMultiplyFn(Object multiplyFn) {
        this.multiplyFn = multiplyFn;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("subFn")
    public Object getSubFn() {
        return this.subFn;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("subFn")
    public void setSubFn(Object subFn) {
        this.subFn = subFn;
    }

    @com.fasterxml.jackson.annotation.JsonGetter("sumFn")
    public Object getSumFn() {
        return this.sumFn;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("sumFn")
    public void setSumFn(Object sumFn) {
        this.sumFn = sumFn;
    }

    @Override
    public boolean equals(Object o) {
        return equalTo(o);
    }

    @Override
    public int hashCode() {
        return hash();
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: dmn-test-cases/standard/tck/1.1/cl3/0031-user-defined-functions/translator/expected/kotlin/dmn/type/TFnLibraryImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "tFnLibrary"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class TFnLibraryImpl : TFnLibrary {
    @get:com.fasterxml.jackson.annotation.JsonGetter("sumFn")
    @set:com.fasterxml.jackson.annotation.JsonGetter("sumFn")
    override var sumFn: kotlin.Any? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("subFn")
    @set:com.fasterxml.jackson.annotation.JsonGetter("subFn")
    override var subFn: kotlin.Any? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("multiplyFn")
    @set:com.fasterxml.jackson.annotation.JsonGetter("multiplyFn")
    override var multiplyFn: kotlin.Any? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("divideFn")
    @set:com.fasterxml.jackson.annotation.JsonGetter("divideFn")
    override var divideFn: kotlin.Any? = null

    constructor() {
    }

    constructor (divideFn: kotlin.Any?, multiplyFn: kotlin.Any?, subFn: kotlin.Any?, sumFn: kotlin.Any?) {
        this.divideFn = divideFn
        this.multiplyFn = multiplyFn
        this.subFn = subFn
        this.sumFn = sumFn
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}
